setup-repo.sh

```diff
@@ -42,7 +42,7 @@ function ensure_hook_is_installed() {
     echo "#!/usr/bin/env bash" >> $pre_commit_file
     echo "current_repo_path=\$(git rev-parse --show-toplevel)" >> $pre_commit_file
     echo "repo_to_format=\"$repo_path\"" >> $pre_commit_file
-    echo 'if [ "$current_repo_path" == "$repo_to_format" ]'" && [ -e \"$DIR\"/format-objc-hook ]; then \"$DIR\"/format-objc-hook; fi" >> $pre_commit_file
+    echo 'if [ "$current_repo_path" == "$repo_to_format" ]'" && [ -e \"$DIR\"/format-objc-hook ]; then \"$DIR\"/format-objc-hook || exit 1; fi" >> $pre_commit_file
```

Comment:
I agree that `set -e` would be preferable, but wanted to defer to the environments and setups of devs.

The reason I didn't use `set -e` was because I wanted to respect pre-existing settings of the precommit hook. This is in case someone is doing other work in their hook that could return an error code but they don't intend for it to stop their script in its tracks. This was the least invasive way to get the script to halt without global changes.